### PR TITLE
Classify 'u8' in Utf8 string literals as keyword

### LINKED ISSUE #63139: Use keyword color for u8 suffix for UTF8 literals
## Summary

The new UTF8 literals use the `u8` suffix which is currently in orange, just like the strings. This makes it extremely hard to notice it while glancing over code, as it very easily gets confused with the rest of the literal. Could we use either the keyword color, or at least some color that's not just the same identical orange as the rest of the literal? Keyword color makes sense given it's a special identifier?

### PR DESCRIPTION
**Pull request by Youssef1313**
image

Fixes #63139

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Should we do this for all literal suffixes?  (i would not do that in this PR, but it's a good question).  It might be a good idea so that the `L` or `l` in a long literal is clearer.

File: src/Workspaces/CSharp/Portable/Classification/Worker.cs [hunk 1]
```diff
@@ -104,6 +106,10 @@ private void ClassifyToken(SyntaxToken token)
                     {
                         AddClassification(span, ClassificationTypeNames.StaticSymbol);
                     }
+                    else if (token.IsKind(SyntaxKind.Utf8StringLiteralToken, SyntaxKind.Utf8SingleLineRawStringLiteralToken, SyntaxKind.Utf8MultiLineRawStringLiteralToken) && token.Text.EndsWith("u8", StringComparison.OrdinalIgnoreCase))
+                    {
+                        AddClassification(new TextSpan(token.Span.End - 2, 2), ClassificationTypeNames.Keyword);
+                    }
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Classification/Worker.cs:112]: what about utf8 interpolated strings?  Or is that not a thing?

File: src/Workspaces/CSharp/Portable/Classification/Worker.cs [hunk 2]
```diff
@@ -104,6 +106,10 @@ private void ClassifyToken(SyntaxToken token)
                     {
                         AddClassification(span, ClassificationTypeNames.StaticSymbol);
                     }
+                    else if (token.IsKind(SyntaxKind.Utf8StringLiteralToken, SyntaxKind.Utf8SingleLineRawStringLiteralToken, SyntaxKind.Utf8MultiLineRawStringLiteralToken) && token.Text.EndsWith("u8", StringComparison.OrdinalIgnoreCase))
+                    {
+                        AddClassification(new TextSpan(token.Span.End - 2, 2), ClassificationTypeNames.Keyword);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Classification/Worker.cs:111]:
```suggestion
                        AddClassification(new TextSpan(token.Span.End - "u8".Length, "u8".Length), ClassificationTypeNames.Keyword);
```

File: src/Workspaces/CSharp/Portable/Classification/Worker.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Workspaces/CSharp/Portable/Classification/Worker.cs:112]: It's not a thing.

**Comment by Youssef1313** [comment]
> Should we do this for all literal suffixes? (i would not do that in this PR, but it's a good question). It might be a good idea so that the `L` or `l` in a long literal is clearer.

@CyrusNajmabadi That makes sense. Do you want a tracking issue for this question?

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**